addons/account/wizard/setup_wizards.py

```diff
@@ -116,8 +120,13 @@ def _compute_linked_journal_id(self):
             record.linked_journal_id = record.journal_id and record.journal_id[0] or record.default_linked_journal_id()
 
     def default_linked_journal_id(self):
-        default = self.env['account.journal'].search([('type', '=', 'bank'), ('bank_account_id', '=', False)], limit=1)
-        return default[:1].id
+        potential_journal_ids = self.env['account.journal'].search([('type', '=', 'bank'), ('bank_account_id', '=', False)])
+        default = self.env['account.journal']
+        for journal_id in potential_journal_ids:
+            empty_journal_count = self.env['account.move'].search_count([('journal_id', '=', journal_id.id)])
+            if empty_journal_count == 0:
+                default = journal_id
```

Comment:
perform an early break in case you find a match to increase performance :)
Usually, performing a search inside a for is bad performance practice but in this case there should not be hundreds of bank journal without bank account. If we happen to have performance issues later we'll rewrite the method using sql or by adding a compute store field on the journal to know if it has entry or not but this is probably an early optimisation therefore we can leave it like this for now :)